src/core/utils/assets.ts

```diff
@@ -363,3 +363,59 @@ export const createAssetQuery = (
           .join(',')}
     }`;
 };
+
+export const getAssetMetadata = async ({
+  address,
+  provider,
+}: {
+  address: Address;
+  provider: Provider;
+}) => {
+  const contract = await getContract({
+    address,
+    abi: erc20ABI,
+    signerOrProvider: provider,
+  });
+  const [decimals, symbol, name] = await Promise.allSettled([
+    contract.decimals(),
+    contract.symbol(),
+    contract.name(),
+  ]);
+
+  return {
+    decimals: extractFulfilledValue<number>(decimals),
+    symbol: extractFulfilledValue<string>(symbol),
+    name: extractFulfilledValue<string>(name),
+  };
+};
+
+export const getAssetBalance = async ({
+  // parsedAsset,
+  assetAddress,
+  currentAddress,
+  // currency,
+  provider,
+}: {
+  // parsedAsset: ParsedUserAsset;
+  assetAddress: Address;
+  currentAddress: Address;
+  // currency: SupportedCurrencyKey;
```

Comment:
why are all these commented?

Reply:
will remove, the method started with that and then i removed it